Snowflake.Data.Tests/IntegrationTests/SFDbCommandIT.cs

```diff
@@ -90,6 +92,44 @@ public void TestCancelExecuteAsync()
             }
         }
 
+        [Test]
+        public void TestExecuteAsyncWithMaxRetryReached()
+        {
+            var mockRestRequester = new MockRetryUntilRestTimeoutRestRequester()
+            {
+                _forceTimeoutForNonLoginRequestsOnly = true
+            };
+
+            using (DbConnection conn = new MockSnowflakeDbConnection(mockRestRequester))
+            {
+                string maxRetryConnStr = ConnectionString + "maxHttpRetries=5";
+
+                conn.ConnectionString = maxRetryConnStr;
+                conn.Open();
+
+                Stopwatch stopwatch = Stopwatch.StartNew();
+                try
+                {
+                    using (DbCommand command = conn.CreateCommand())
+                    {
+                        command.CommandText = "select 1;";
+                        Task<object> t = command.ExecuteScalarAsync();
+                        t.Wait();
+                    }
+                    Assert.Fail();
+                }
+                catch (Exception e)
+                {
+                    Assert.IsInstanceOf<TaskCanceledException>(e.InnerException);
+                }
+                stopwatch.Stop();
+
+                // retry 5 times with backoff 1, 2, 4, 8, 16 seconds
+                // but should not delay more than another 16 seconds
+                Assert.Less(stopwatch.ElapsedMilliseconds, 51 * 1000);
+                Assert.GreaterOrEqual(stopwatch.ElapsedMilliseconds, 30 * 1000);
```

Comment:
Are you sure 30s isn't too much?

Reply:
That's the number used for the original retry test case: https://github.com/snowflakedb/snowflake-connector-net/pull/803/files#diff-1bb87bf600c092adcc18c664d65701025fe5a30d0d99722aa1fbc3d1e3c9836aL437

This is for non-login requests which doesn't use the new retry strategy and still uses the original back off pattern of 1, 2, 4, 8, and 16